src/tools/lint-docs/src/lib.rs

```diff
@@ -141,12 +141,11 @@ impl<'a> LintExtractor<'a> {
             let (doc, name) = loop {
                 match lines.next() {
                     Some((lineno, line)) => {
-                        let line = line.trim();
-                        if let Some(text) = line.strip_prefix("/// ") {
-                            doc_lines.push(text.trim().to_string());
-                        } else if line.starts_with("///") {
+                        if let Some(text) = line.trim_start().strip_prefix("/// ") {
+                            doc_lines.push(text.to_string());
+                        } else if line.trim_start().starts_with("///") {
                             doc_lines.push("".to_string());
```

Comment:
I don't think `///a` should push an empty line.

Reply:
Yeah, it does seem weird. @ehuss: did you do this intentionally or was it a mistake?

Reply:
Maybe something like
```rust
if let Some(text) = line.trim_start().strip_prefix("///") {
    doc_lines.push(text.trim_start().to_string());
}

Reply:
No, I think that will cause the same issue that this is trying to fix.